src/XIVLauncher.Core/Components/MainPage/MainPage.cs

```diff
@@ -183,8 +183,8 @@ public async Task<bool> Login(string username, string password, bool isOtp, bool
             return false;
 
         var otp = string.Empty;
-
-        if (isOtp && !App.UniqueIdCache.HasValidCache(username))
+        
+        if (isOtp && (!App.UniqueIdCache.HasValidCache(username) || App.Settings.IsUidCacheEnabled == false))
```

Comment:
We could do it this way instead, if we want to delete the uid cache.
```suggestion
        if (isOtp && App.UniqueIdCache.HasValidCache(username) && App.Settings.IsUidCacheEnabled == false)
            Program.ResetUIDCache();

        if (isOtp && !App.UniqueIdCache.HasValidCache(username))
```